### PR TITLE
Add media selection

### PR DESCRIPTION
**Pull request by trickreich**

### CODE REVIEW

File: Controller/Product/ProductController.php [hunk 1]
```diff
@@ -38,6 +39,19 @@ public function __construct(MessageBusInterface $messageBus, ViewHandlerInterfac
         $this->setViewHandler($viewHandler);
     }
 
+    public function putAction(Request $request, string $id): Response
+    {
+        $locale = $request->query->get('locale');
+
+        $this->messageBus->dispatch(
```
Comment (wachterjohannes) [inline Controller/Product/ProductController.php]: should fit into a single line

File: Model/Product/Handler/Message/ModifyProductMessageHandler.php [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                $media = $this->mediaRepository->find($mediaReferenceValueObject->getMediaId());
+                $mediaReference = $this->productMediaReferenceRepository->create(
+                    $product,
+                    $media,
+                    $mediaReferenceValueObject->getType()
+                );
+            }
+
+            if ($mediaReference->getSyliusId()) {
+                $mediaReference->setActive($mediaReferenceValueObject->getActive());
```
Comment (wachterjohannes) [inline Model/Product/Handler/Message/ModifyProductMessageHandler.php]: can this be changed? i think this should always be active? therefor you can remove this part of the if

File: Model/Product/Handler/Message/ModifyProductMessageHandler.php [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+    public function __invoke(ModifyProductMessage $message): ProductInterface
+    {
+        $product = $this->productRepository->findById($message->getId());
+        if (!$product) {
+            throw new ProductNotFoundException($message->getId());
+        }
+
+        $sorting = 1;
```
Comment (wachterjohannes) [inline Model/Product/Handler/Message/ModifyProductMessageHandler.php]: why not using the index of the array?

File: Model/Product/Message/ValueObject/ProductImageValueObject.php [hunk 4]
```diff
@@ -31,7 +31,9 @@ public function getId(): int
 
     public function getType(): string
     {
-        return $this->getStringValueWithDefault('type');
+        $typo = $this->getNullableStringValue('type');
```
Comment (wachterjohannes) [inline Model/Product/Message/ValueObject/ProductImageValueObject.php]: `type`

File: Model/Product/Handler/Message/ModifyProductMessageHandler.php (hunk 2 above)
Comment (trickreich) [reply Model/Product/Handler/Message/ModifyProductMessageHandler.php]: It's ok this way

**Pull request merged**